test/Benchmarks/src/Vector/Distinct.enso

```diff
@@ -1,27 +1,33 @@
 from Standard.Base import all
 
-from Standard.Test import Bench
+from Standard.Test import Bench, Phase_Conf
 
 import project.Vector.Utils
 
+type Setup
+    Value random_vec uniform_vec random_text_vec uniform_text_vec
```

Comment:
Actually everything seems to work _without lazy atom fields_. Even if I keep:
```ruby
Value random_vec uniform_vec random_text_vec uniform_text_vec
```
the value of `uniform_text_vec` is computed just once. Moreover there is no need to call the getters to initialize the values in such case.

Reply:
Oh, no but doesn't that undo the whole point of setup logic?

If I understand correctly, isn't this setup logic now run not only when running benchmarks but also **when gathering** them??

Given that the benchmarks may prepare quite big vectors, this may significantly slow-down the gathering phase.

Or am I reading something wrong?

Reply:
> If I understand correctly, isn't this setup logic now run not only when running benchmarks but also when gathering them??

That's correct. Moreover, even simply prepending `~` to the field names does not fix this problem. When you have:
```
type Data
  Value ~random_vec

setup =
  random_vec = Utils.make_random_vec 1000000
  Data.Value random_vec

collect_benches = Bench.build builder->
  data = setup
  builder.group "My_Group" options group_builder->
    group_builder.specify "My_Bench" <|
      data.random_vec.distinct
```
the `Utils.make_random_vec 1000000` expression is evaluated both during **benchmark gathering** (which is not OK) and during the first warmup (which is OK). There are two possible solutions for this, none of which are particularly nice:

### 1)
Initialize `Data.Value` with the expression rather than with `random_vec`:
```
setup = 
  Data.Value (Utils.make_random_vec 1000000)
```
This solution gets unreadable once there are many fields in `Data`, as all the expressions must be specfied on the same line.

### 2)
Introduce a separate method for each field:
```
setup =
  Data.Value create_random_vec

create_random_vec =
  Utils.make_random_vec 1000000
```
Which is more readable, but very verbose.

---------------------------------------
I will delay addressing this issue once the **gathering benchmarks** step starts to be too slow.

Reply:
Resolved by introducing *lazy data setup* like in https://github.com/enso-org/enso/pull/7519/commits/3ca6cd79c30655790fde40b9fb682d258edfaae8